app/lib/features/events/pages/event_details_page.dart

```diff
@@ -500,24 +493,33 @@ class _EventDetailPageConsumerState extends ConsumerState<EventDetailPage> {
   }
 
   Widget _buildEventDataSet(CalendarEvent ev) {
-    final inDays =
+    final agoTime =
         Jiffy.parseFromDateTime(toDartDatetime(ev.utcStart()).toLocal())
-            .endOf(Unit.day)
+            .endOf(Unit.hour)
             .fromNow();
     final startDate =
         Jiffy.parseFromDateTime(toDartDatetime(ev.utcStart()).toLocal())
-            .format(pattern: 'EEE, MMM dd AT hh:mm');
+            .format(pattern: 'EEE, MMM dd, yyyy AT hh:mm');
     final endDate =
         Jiffy.parseFromDateTime(toDartDatetime(ev.utcEnd()).toLocal())
-            .format(pattern: 'EEE, MMM dd AT hh:mm');
+            .format(pattern: 'EEE, MMM dd, yyyy AT hh:mm');
+
+    String eventTimingTitle = '';
+    if (getEventType(ev) == EventFilters.ongoing) {
+      eventTimingTitle = '${L10n.of(context).started} $agoTime';
+    } else if (getEventType(ev) == EventFilters.upcoming) {
+      eventTimingTitle = '${L10n.of(context).starts} $agoTime';
+    } else if (getEventType(ev) == EventFilters.past) {
+      eventTimingTitle = '${L10n.of(context).ended} $agoTime';
+    }
```

Comment:
this screams for the new [dart switch expression](https://dart.dev/language/branches#switch-expressions) ;) 
```suggestion
    String eventTimingTitle = switch(getEventType(ev)) {
      EventFilters.ongoing =>'${L10n.of(context).started} $agoTime',
      EventFilters.upcoming => '${L10n.of(context).starts} $agoTime',
      EventFilters.past => '${L10n.of(context).ended} $agoTime',
    };
```